### ISSUE TITLE
Domain arcadeblogger.com is opened in Google container but it shouldn't

### ISSUE THREAD

**Issue by mastazi** [containers-everywhere/contain-google#81]
The domain arcadeblogger.com is not a Google domain. It is not a commercial venture at all and it is maintained by an individual as you can see here https://arcadeblogger.com/about/

This issue is possibly related to issue #74 - because in both cases the end of the domain name is the same as the name of a Google-owned domain. It seems that there is some sort of wildcard such as 

    *blogger.com
    *google.com

when instead it should be:

    *.blogger.com
    *.google.com

(notice the dot after the wildcard in the last two examples).

**Comment by B00ze64** [comment]
The code looks fine but there might be bugs:

```js
function generateGoogleHostREs () {
  const matchOperatorsRegex = /[|\\{}()[\]^$+*?.-]/g;
  for (let googleDomain of GOOGLE_DOMAINS) {
    googleDomain = googleDomain.replace(matchOperatorsRegex, '\\$&');
    googleHostREs.push(new RegExp(`(^|\.)${googleDomain}$`));
  } ...
```
The "(^|\\.)" basically means at start (e.g. blogger.com) or after a DOT (e.g. test.blogger.com). Must be a bug somewhere; it's clearly not working as expected. The current Facebook Container uses a slightly different pattern:

```js
function generateFacebookHostREs () {
  for (let facebookDomain of FACEBOOK_DOMAINS) {
    facebookHostREs.push(new RegExp(`^(.*\\.)?${facebookDomain}$`));
  } }
```
Facebook Container was updated 2 months ago, maybe they fixed the RegExp? Google Container has not been touched in a year, @Perflyst has not been here in some time - The issue you report has been reported multiple times (#68 #74 ...)

It is quite easy to replace the 1 line in the code with the version from the newer Facebook Container to see if that fixes it, however, Firefox's "All Addons must be Signed" prevents me from doing that unless I install Dev edition (or something like that). So annoying.

**Comment by B00ze64** [comment]
Actually I've just looked at FBC commits and I do not see them changing the RegExp in that function recently, so the bug is probably somewhere else. They are constantly editing the background script. And there are ~180 bugs opened for FBC alone, no wonder Google Container has some too!

**Comment by Perflyst** [comment]
> Google Container has not been touched in a year

If anyone has interested in maintaining it further feel free to contact me.

**Comment by B00ze64** [comment]
Sorry @Perflyst did not mean to sound annoyed or ungrateful. If we look at the container "market" we see tons of ancillary addOns that were created and are unable to keep-up with the official ones (MAC and FBC). Containers are a bit complicated when multiple addOns manage them, and Mozilla keeps updating their own addOns trying to make it all work, it's a job to try and keep-up!

**Comment by m-a-orlowski** [comment]
I think it's just lacking one backslash when creating new RegExp object.
So it should look like this:

``googleHostREs.push(new RegExp(`(^|\\.)${googleDomain}$`));``
and
``youtubeHostREs.push(new RegExp(`(^|\\.)${youtubeDomain}$`));``

That should hopefully be enough to fix this issue.

**Comment by Perflyst** [comment]
Closed with #89
